### PR TITLE
Optimization for Statistics 

### PR DESCRIPTION
**Pull request by BohdanBuhrii**

### CODE REVIEW

File: TransIT.BLL/DTOs/StatisticsDTO.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+
+namespace TransIT.BLL.DTOs
+{
+    public class StatisticsDTO
+    {
+        public string FieldName { get; set; }
+
+        public ICollection<int> Statistics { get; set; }
```
Comment (igolovchuk) [inline TransIT.BLL/DTOs/StatisticsDTO.cs]: Use IEnumerable instead ICollection

File: TransIT.BLL/Services/ImplementedServices/StatisticsService.cs [hunk 2]
```diff
@@ -101,5 +102,76 @@ public async Task<List<int>> GetMalfunctionGroupStatistics(string malfunctionGro
 
             return result;
         }
+
+        public async Task<List<StatisticsDTO>> GetAllGroupsStatistics()
+        {
```
Comment (igolovchuk) [inline TransIT.BLL/Services/ImplementedServices/StatisticsService.cs:107]: try to rewrite all creating new lists logic to using iterator(yield), add 'Async' suffix to the method names that use async code

for ex:

```
 public async Task<IEnumerable<StatisticsDTO>> GetAllGroupsStatisticsAsync()
{
var malfunctionGroups = await _unitOfWork.MalfunctionGroupRepository.GetAllAsync();

         foreach (MalfunctionGroup group in malfunctionGroups)
        {
            yield return new StatisticsDTO
            {
                FieldName = group.Name,
                Statistics = await GetMalfunctionGroupStatistics(group.Name)
            };
        }
    }
```

**Review by igolovchuk — CHANGES_REQUESTED** [review]
please read comments in files

**Review by andriyshevchenko — DISMISSED** [review]

File: TransIT.BLL/Services/ImplementedServices/StatisticsService.cs (hunk 2 above)
Comment (BohdanBuhrii) [reply TransIT.BLL/Services/ImplementedServices/StatisticsService.cs:107]: As I found, to use this way we need to replace Task<IEnumerable<StatisticsDTO>> with IAsyncEnumerable<StatisticsDTO>, which now available only in "preview" language version. Is this ok?

Comment (igolovchuk) [reply TransIT.BLL/Services/ImplementedServices/StatisticsService.cs:107]: ok lets leave as is, I thought you have the latest SDK

**Review by igolovchuk — APPROVED** [review]

**Review by andriyshevchenko — APPROVED** [review]

**Review by Vladacdc — APPROVED** [review]

**Pull request merged**